inc/class-my-photon.php

```diff
@@ -366,13 +366,15 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 						$photon_args['w'] = $image_args['width'];
 				} else {
 					if( 'resize' == $transform ) {
-						// Lets make sure that we don't upscale images since wp never upscales them as well
+						// Lets make sure that we don't upscale images since wp never upscales them as well.
 						$image_meta = wp_get_attachment_metadata( $attachment_id );
 
-						$smaller_width  = ( ( $image_meta['width']  < $image_args['width']  ) ? $image_meta['width']  : $image_args['width']  );
-						$smaller_height = ( ( $image_meta['height'] < $image_args['height'] ) ? $image_meta['height'] : $image_args['height'] );
+						if ( is_array( $image_meta ) ) {
```

Comment:
This will fix the offset issue, but will result in an unexpected result, since the the 'resize' transformation expects to size an image to fit in a specific location. I think it would be rare that image meta won't exist at all for an attachment, but if so, perhaps it's best to go ahead and respect the `image_args` passed here rather than throw them out if no meta is found?

Reply:
This original code was forked from jetpack. Might be helpful to compare [the latest updates there](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/class.photon.php#L765-L785).

Generally speaking, this class could probably benefit from a massive update against the latest jetpack code.

Reply:
Yes! Pull request updated!

Reply:
I forgot that this was meant to be a recreation of what Jetpack would do in this scenario, thank @mboynes. It does look like updating this code to match what Jetpack is doing in the snippet that Matt linked to would be best. Could you give that a go, @renatonascalves?

Reply:
Sure! :)